src/shared/telemetry/telemetryUtils.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as vscode from 'vscode'
+import { ext } from '../extensionGlobals'
+import { Datum, ResultWithTelemetry } from './telemetryEvent'
+
+export function defaultMetricDatum(name: string): Datum {
+    return {
+        name: name,
+        unit: 'Count',
+        value: 1
+    }
+}
+
+export function registerCommand<T>(
+    command: string,
+    callback: (...args: any[]) => (Promise<ResultWithTelemetry<T> | void>),
+    thisArg?: any
+): vscode.Disposable {
+    return vscode.commands.registerCommand(
```

Comment:
For testability, add the following parameter to `registerCommand<T>`:

```
export function registerCommand<T>(
    ...
    register: typeof vscode.commands.registerCommand = vscode.commands.registerCommand
```

Reply:
Once you add this, can you add tests? They should validate that `telemetry.record` is called with the expected arguments (for `duration` and `createTime`, just verify that they're truthy).

Reply:
Good catch!

Reply:
:+1: